core/domain/base_model_validators.py

```diff
@@ -84,12 +88,57 @@ def __init__(
             model_class: ClassObject. The external model class.
             model_ids: list(str). The list of external model ids to fetch the
                 external models.
+
+        Raises:
+            Exception. This class was used instead of using the
+                UserSettingsModelFetcherDetails for the UserSettingsModel.
         """
+        if model_class == user_models.UserSettingsModel:
+            raise Exception(
+                'Please use the UserSettingsModelFetcherDetails ' +
+                'for UserSettingsModel')
         self.field_name = field_name
         self.model_class = model_class
         self.model_ids = model_ids
 
 
+class UserSettingsModelFetcherDetails(python_utils.OBJECT):
+    """Value object providing the class and ids to fetch the
```

Comment:
Isn't the class already implicit in the name? Why does it need to be "provided" again?

Please write your docstrings more carefully.

Reply:
Done.